src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/SendNotificationHandlerTest.java

```diff
@@ -0,0 +1,114 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.DispatchPriority;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.service.NotificationSender;
+
+@RunWith(MockitoJUnitRunner.class)
+@SuppressWarnings("unchecked")
+public class SendNotificationHandlerTest {
+
+    @Mock private NotificationSender<AsylumCase> notificationSender;
+    @Mock private Callback<AsylumCase> callback;
+
+    private SendNotificationHandler sendNotificationHandler;
+
+    @Before
+    public void setUp() {
+
+        sendNotificationHandler =
+            new SendNotificationHandler(notificationSender);
+    }
+
+    @Test
+    public void should_send_notification_and_update_the_case() {
```

Comment:
Should we be able to send notifications even if the same notification has been sent already? And if so, would we want to cover that in a test?

Reply:
It is a good thought.

Currently we prevent duplicate emails within a short time (60 seconds) within notifications API, but not over the entire lifecycle of the case.

There maybe cases where the same email may get sent in the future, I am thinking when a case is put back into a previous state for correction. When the case progresses again, another email will likely be needed for the same action taken the second time.

Reply:
yeah that would make sense to me, or if there had been some issue and certain notifications needed to be resent. So would it be worth adding that test here, or is that covered by the `ia-case-notifications-api` code?

Reply:
I think I may have found the answer to my own question here: https://github.com/hmcts/ia-case-notifications-api/blob/8653e8da0b41be6ac673244c3eaabc3b053080a0/src/test/java/uk/gov/hmcts/reform/iacasenotificationsapi/infrastructure/clients/GovNotifyNotificationSenderTest.java#L70